pilot/pkg/model/proxy_config.go

```diff
@@ -140,6 +141,9 @@ func mergeWithPrecedence(pcs ...*meshconfig.ProxyConfig) *meshconfig.ProxyConfig
 		if pcs[i].GetImage() != nil {
 			merged.Image = pcs[i].GetImage().DeepCopy()
 		}
+		if pcs[i].GetTracing() != nil {
```

Comment:
This seems a bit suspicious to me. mergeWithPrecedence was added for the ProxyConfig CRD, so I am a surprised to see non-CRD related code here. Maybe its correct though, let me look into the surrounding calls a bit more